manage/ships/repair.php

```diff
@@ -8,7 +8,10 @@
 
 $container = new Container($configuration);
 $shipLoader = $container->getShipLoader();
-
-$shipLoader->repairShip($id);
+$ship = $shipLoader->findOneById($id);
+if ($ship === null) {
+    header('Location: /manage/ships/view.php');
```

Comment:
Did you test this logic? This is trying to redirect to the view page with no id. Also, I think it will fall through this conditional because there is no `die;`. So then it will call the `repairShip` method below.

I think it would be better to just display an error message here anyway.